src/alire/alire-properties-licenses.adb

```diff
@@ -1,5 +1,84 @@
+with Alire.Errors;
+
 package body Alire.Properties.Licenses is
 
+   function Legacy_To_SPDX (Legacy : Licensing.Licenses) return String
+   is (case Legacy is
+          when AFL_3_0            => "AFL-3.0",
+          when AGPL_3_0           => "AGPL-3.0-only",
+          when Apache_2_0         => "Apache-2.0",
+          when Artistic_2_0       => "Artistic-2.0",
+          when BSD_2_Clause       => "BSD-2-Clause",
+          when BSD_3_Clause_Clear => "BSD-3-Clause-Clear",
+          when BSD_3_Clause       => "BSD-3-Clause",
+          when BSL_1_0            => "BSL-1.0",
+          when CC0_1_0            => "CC0-1.0",
+          when CC_BY_4_0          => "CC-BY-4.0",
+          when CC_BY_SA_4_0       => "CC-BY-SA-4.0",
+          when ECL_2_0            => "ECL-2.0",
+          when EPL_1_0            => "EPL-1.0",
+          when EPL_2_0            => "EPL-2.0",
+          when EUPL_1_1           => "EUPL-1.1",
+          when EUPL_1_2           => "EUPL-1.2",
+          when GPL_2_0            => "GPL-2.0-only",
+          when GPL_3_0            => "GPL-3.0-only",
+          when ISC                => "ISC",
+          when LGPL_2_1           => "LGPL-2.1-only",
+          when LGPL_3_0           => "LGPL-3.0-only",
+          when LPPL_1_3c          => "LPPL-1.3c",
+          when MIT                => "MIT",
+          when MPL_2_0            => "MPL-2.0",
+          when MS_PL              => "MS-PL",
+          when MS_RL              => "MS-RL",
+          when NCSA               => "NCSA",
+          when OFL_1_1            => "OFL-1.1",
+          when OSL_3_0            => "OSL-3.0",
+          when PostgreSQL         => "PostgreSQL",
+          when Unlicense          => "Unlicense",
+          when WTFPL              => "WTFPL",
+          when Zlib               => "Zlib",
+          when GMGPL_2_0          => "GPL-2.0-only WITH GCC-exception-2.0",
+          when GMGPL_3_0          => "GPL-3.0-only WITH GCC-exception-3.1",
+          when Public_Domain | Custom | Unknown =>
+             raise Checked_Error
+               with Errors.Set ("Invalid legacy license conversion for '" &
+                                  Legacy'Img & "'"));
```

Comment:
For my edification: does this mean that SPDX does not support custom licenses? (so we lose this capability?)

Reply:
Yes SPDX doesn't allow license outside of the list they provide. And that's going to be a problem for the existing index.

I think there are several options:
 - `Custom` is not allowed anymore and the `licenses` must only be used for SPDX licenses.
 - Add an option to accept `custom` in the SPDX library.
 - Add a detection of `custom` in `Alire` to bypass the SPDX library.

Reply:
I will try to add a `other-<License-name-here>` support in the SPDX library.